### PR TITLE
create delete endpoint

### PR DESCRIPTION
**Pull request by MichaelJacob21012**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/EM-6177

### Change description ###
create delete endpoint

### CODE REVIEW

**Review by yogesh-hullatti — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/HearingRecordingServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        this.blobStorageDeleteService = blobStorageDeleteService;
+    }
+
+    @Override
+    public void deleteCaseHearingRecordings(Collection<Long> ccdCaseIds) {
+        List<HearingRecording> hearingRecordings = hearingRecordingRepository.deleteByCcdCaseIdIn(ccdCaseIds);
+        List<HearingRecordingSegment> segments =
+            hearingRecordings.stream().flatMap(hearingRecording -> hearingRecording.getSegments().stream()).toList();
+
+        segments.forEach(segment -> blobStorageDeleteService.deleteBlob(
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/HearingRecordingServiceImpl.java:32]: we should first delete the blobs and then delete from DB?

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/HearingRecordingServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import uk.gov.hmcts.reform.em.hrs.domain.HearingRecordingSegment;
+import uk.gov.hmcts.reform.em.hrs.dto.HearingSource;
+import uk.gov.hmcts.reform.em.hrs.repository.HearingRecordingRepository;
+
+import java.util.Collection;
+import java.util.List;
+
+@Service
+@Transactional
+public class HearingRecordingServiceImpl implements HearingRecordingService {
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/HearingRecordingServiceImpl.java:15]: another name would better like HearingRecordingDeleteService

File: src/cftlib/java/uk/gov/hmcts/reform/em/hrs/cftlib/CftLibConfig.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public void configure(CFTLib lib) throws Exception {
         lib.configureRoleAssignments(json);
 
         lib.importDefinition(Files.readAllBytes(
-            Path.of("src/functionalTest/resources/CCD_HRS_v1.1-AAT.xlsx")));
+            Path.of("src/functionalTest/resources/CCD_HRS_v1.2-AAT.xlsx")));
```
Comment (mdayican) [inline src/cftlib/java/uk/gov/hmcts/reform/em/hrs/cftlib/CftLibConfig.java:42]: unrelated changes like this one and other renaming stuff can go separately, so if we break something we will revert just the related change and reviewing would easy.

File: src/main/java/uk/gov/hmcts/reform/em/hrs/controller/HearingRecordingController.java [hunk 4]
```diff
@@ -290,6 +298,32 @@ public ResponseEntity<Void> getSegmentBinaryForSharee(
...
+                description = "Authorization (Idam Bearer token)", required = true,
+                schema = @Schema(type = "string"))})
+    @ApiResponses(
+        value = {@ApiResponse(responseCode = "204", description = "Successful deletion"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "403", description = "Forbidden")}
+    )
+    public ResponseEntity<Long> deleteCaseHearingRecordings(@RequestBody final List<Long> ccdCaseIds) {
+        if (!deleteCaseEndpointEnabled) {
+            return ResponseEntity.notFound().build();
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/controller/HearingRecordingController.java:321]: not found or 403 forbidden ?

File: src/main/java/uk/gov/hmcts/reform/em/hrs/domain/HearingRecording.java [hunk 5]
```diff
@@ -81,6 +81,9 @@ public class HearingRecording {
     @OneToMany(cascade = CascadeType.ALL, mappedBy = "hearingRecording")
     private Set<HearingRecordingSegment> segments;
 
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "hearingRecording")
+    private Set<HearingRecordingSharee> sharees;
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/domain/HearingRecording.java:85]: this will have performance issue on retrieval of records and even on delete, you can find other way to delete all

Comment (MichaelJacob21012) [reply src/main/java/uk/gov/hmcts/reform/em/hrs/domain/HearingRecording.java:85]: have investigated behaviour on call with @mdayican using demo environment
looks ok as is bidirectional with lazy fetch type for the mapped many to one

**Review by yogesh-hullatti — APPROVED** [review]

**Review by mdayican — APPROVED** [review]
